lhotse/dataset/input_strategies.py

```diff
@@ -0,0 +1,138 @@
+from typing import Dict, Tuple
+
+import torch
+
+from lhotse import CutSet, FeatureExtractor
+from lhotse.dataset.collation import collate_audio, collate_features
+from lhotse.utils import compute_num_frames, supervision_to_frames, supervision_to_samples
+
+
+class InputStrategy:
+    """
+    Converts a :class:`CutSet` into a collated batch of audio representations.
+    These representations can be e.g. audio samples or features.
+    They might also be single or multi channel.
+
+    This is a base class that only defines the interface.
+    """
+
+    def __call__(self, cuts: CutSet) -> Tuple[torch.Tensor, torch.IntTensor]:
+        raise NotImplementedError()
+
+    def supervision_intervals(self, cuts: CutSet) -> Dict[str, torch.Tensor]:
+        """
+        Returns a dict that specifies the start and end bounds for each supervision,
+        as a 1-D int tensor.
+
```

Comment:
Would be nice to have more clarity on the string and the tensor in terms of what they represent.  E.g. how many elements does the tensor have, and what is its meaning?